fe/fe-core/src/main/java/com/starrocks/system/TabletComputeNodeMapper.java

```diff
@@ -192,26 +202,61 @@ public List<Long> computeNodesForTablet(Long tabletId) {
     }
 
     public List<Long> computeNodesForTablet(Long tabletId, int count) {
-        String thisResourceIsolationGroup = GlobalStateMgr.getCurrentState().getNodeMgr().getMySelf().
-                getResourceIsolationGroup();
+        String thisResourceIsolationGroup = GlobalStateMgr.getCurrentState().getNodeMgr().getMySelf().getResourceIsolationGroup();
         return computeNodesForTablet(tabletId, count, thisResourceIsolationGroup);
     }
 
     public List<Long> computeNodesForTablet(Long tabletId, int count, String resourceIsolationGroup) {
         readLock.lock();
         try {
             if (!this.resourceIsolationGroupToTabletMapping.containsKey(resourceIsolationGroup)) {
-                LOG.warn(String.format("Requesting node for resource isolation group %s, to which"
-                        + " there is not a known CN assigned.", resourceIsolationGroup));
+                LOG.warn(String.format(
+                        "Requesting node for resource isolation group %s, to which" + " there is not a known CN assigned.",
+                        resourceIsolationGroup));
                 return Collections.emptyList();
             }
             TabletMap m = this.resourceIsolationGroupToTabletMapping.get(resourceIsolationGroup);
-            return m.tabletToComputeNodeId.get(tabletId, count);
+            List<Long> computeNodes = m.tabletToComputeNodeId.get(tabletId, count);
+            // Update tracking information
+            tabletMappingCount.computeIfAbsent(tabletId, k -> new AtomicLong()).incrementAndGet();
+            computeNodes.forEach(
+                    nodeId -> computeNodeReturnCount.computeIfAbsent(nodeId, k -> new AtomicLong()).incrementAndGet());
+            return computeNodes;
         } finally {
             readLock.unlock();
         }
     }
 
+    // Number of times that this mapper has returned the given compute node as the return value for `computeNodesForTablet`.
+    public Long getComputeNodeReturnCount(Long computeNodeId) {
+        return computeNodeReturnCount.getOrDefault(computeNodeId, new AtomicLong(0)).get();
+    }
+
+    // Returns the tablet mapped to the number of times which some caller has requested the appropriate CN.
+    // Only reports the number of times since this particular FE came up.
+    public ConcurrentMap<Long, AtomicLong> getTabletMappingCount() {
+        return tabletMappingCount;
+    }
+
+    // Key is compute node id. Value is number of distinct tablets for which this mapper instance has returned the given CN.
```

Comment:
Leaving a retroactive comment from our discussion.

The caveat here is that if the queries are skewed for the most recent days then we'll see some CNs report counts where others will be zeroed out correct? But the benefit still is that observability is better than no observability. And to your point it will enable us to answer questions like "is this CN really hot because it's handling too many tablets or because some tablet is really hot right now". If we don't periodically clear the scanned count we may end up losing signal over time if we don't restart CN nodes.